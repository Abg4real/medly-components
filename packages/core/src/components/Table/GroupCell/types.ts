import { GridTemplateProps } from '../types';
export interface Props extends GridTemplateProps {
    hide: boolean;
}
