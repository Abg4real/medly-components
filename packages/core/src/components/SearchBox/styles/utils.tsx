import { SvgIcon } from '@medly-components/icons';
import { css } from '@medly-components/utils';
import { Props } from '../types';

export const getSearchBoxSize = ({ theme, size }: Props) => theme.searchBox.sizes[size].height;

export const getBorderAndBoxShadow = (color: string, boxShadow: string) => {
    return css`
        border-color: ${color};
        box-shadow: ${boxShadow};
    `;
};

export const getIconWrapperStyle = ({ theme, size }: Props) => {
    return css`
        width: ${getSearchBoxSize};
        min-width: ${getSearchBoxSize};
        height: auto;
        display: inline-flex;
        align-items: center;
        justify-content: center;
        ${SvgIcon} {
            cursor: pointer;
            font-size: ${theme.searchBox.sizes[size].iconSize};
        }
    `;
};
