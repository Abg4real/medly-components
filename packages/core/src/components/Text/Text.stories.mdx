import { Text } from './Text';
import { defaultTheme } from '@medly-components/theme';
import { Preview, Story, Meta, Props } from '@storybook/addon-docs/blocks';
import * as stories from './Text.stories';

<Meta title="Core/Text" component={Text} />

# Text

It renders text in `span` html tag when text weight is `light` | `normal` and uses `strong` html tag for rendering text with`strong` text weight.

<Preview withToolbar>
    <Story name="Basic">{stories.Basic}</Story>
</Preview>

### Props

<Props of={Text} />

### TextTheme

<Props of={stories.ThemeInterface} />

### TextTheme Defaults

<code>{JSON.stringify(defaultTheme.text, null, 2)}</code>
