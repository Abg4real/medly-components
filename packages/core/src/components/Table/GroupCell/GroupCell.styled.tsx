import { centerAligned, styled, WithThemeProp } from '@medly-components/utils';
import Text from '../../Text';
import { GridTemplateProps } from '../types';

export const GroupCellStyled = styled('div').attrs(({ gridTemplateColumns }: GridTemplateProps) => ({
    style: {
        gridTemplateColumns
    }
}))<GridTemplateProps>`
    display: grid;
    width: 100%;
    height: 100%;
`;

export const GroupCellTitleStyled = styled(Text.Style)<WithThemeProp>`
    font-size: ${({ theme }) => theme.font.sizes.M3};
    font-weight: ${({ theme }) => theme.font.weights.Strong};
    padding: 5px 0px 5px 10px;
    grid-column: -1/1;
    text-overflow: initial;
    overflow: auto;
    white-space: normal;
`;
