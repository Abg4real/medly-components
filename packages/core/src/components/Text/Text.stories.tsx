import { defaultTheme, FontSizes, FontWeights, TextTheme } from '@medly-components/theme';
import { boolean, color, select } from '@storybook/addon-knobs';
import React from 'react';
import { Text } from './Text';

const sizeOptions: FontSizes[] = ['S1', 'M1', 'M2', 'M3', 'L1', 'L2', 'L3', 'L4'];
const weightOptions: FontWeights[] = ['Light', 'Regular', 'Medium', 'Strong'];

export const ThemeInterface = (props: TextTheme): any => null;
ThemeInterface.defaultProps = {
    ...defaultTheme.text
};

export const Basic = () => (
    <Text
        fullWidth={boolean('Full Width', false)}
        uppercase={boolean('Upper Case', false)}
        lineThrough={boolean('Line through', false)}
        textColor={color('Color', '#012040')}
        textSize={select('Text Size', sizeOptions, 'L1')}
        textWeight={select('Text Weight', weightOptions, 'Regular')}
    >
        Lorem ipsum dolor sit, amet consectetur adipisicing elit.
    </Text>
);
