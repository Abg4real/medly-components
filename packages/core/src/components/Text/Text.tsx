import { WithStyle } from '@medly-components/utils';
import React, { SFC, useMemo } from 'react';
import { TextStyled } from './Text.styled';
import { Props } from './types';

export const Text: SFC<Props> & WithStyle = React.memo(
    React.forwardRef(({ as, textVariant, textWeight, children, ...restProps }, ref) => {
        const componentType = useMemo(() => {
            let type: keyof JSX.IntrinsicElements | React.ComponentType<any> = 'span';

            if (textVariant === 'body1' || textVariant == 'body2') {
                type = textWeight === 'Strong' || textWeight === 'ExtraStrong' ? 'strong' : 'span';
            } else {
                type = textVariant;
            }

            return type;
        }, [textWeight, textVariant]);
        return (
            <TextStyled {...{ textVariant, textWeight, ...restProps }} as={as || componentType} ref={ref}>
                {children}
            </TextStyled>
        );
    })
);

Text.defaultProps = {
    lineThrough: false,
    uppercase: false,
    fullWidth: false,
    textColor: 'inherit'
};
Text.displayName = 'Text';
Text.Style = TextStyled;
