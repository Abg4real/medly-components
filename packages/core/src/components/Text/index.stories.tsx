import { FontSizeTheme, FontWeightTheme } from '@medly-components/theme';
import { boolean, color, select } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React from 'react';
import Text from './Text';

const sizeOptions: Array<keyof FontSizeTheme> = ['S1', 'M1', 'M2', 'M3', 'L1', 'L2', 'L3', 'L4'];
const weightOptions: Array<keyof FontWeightTheme> = ['Light', 'Normal', 'Strong'];

storiesOf('Core', module).add('Text', () => (
    <Text
        uppercase={boolean('Upper Case', false)}
        lineThrough={boolean('Line through', false)}
        textColor={color('Color', '#012040')}
        textSize={select('Text Size', sizeOptions, 'L1')}
        textWeight={select('Text Weight', weightOptions, 'Normal')}
    >
        Lorem ipsum dolor sit, amet consectetur adipisicing elit. Quaerat excepturi quos vitae quisquam hic molestias et cumque nostrum
        corrupti ad dolorem ab laborum iste veniam, animi optio ipsa quod error!
    </Text>
));
