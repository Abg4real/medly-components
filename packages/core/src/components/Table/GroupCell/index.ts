import { GroupCellStyled as GroupCell, GroupCellTitleStyled as GroupCellTitle } from './GroupCell.styled';

export { GroupCell, GroupCellTitle };
