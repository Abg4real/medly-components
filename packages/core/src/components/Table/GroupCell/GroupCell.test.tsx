import { TestUtils } from '@medly-components/utils';
import React from 'react';
import Text from '../../Text';

import { GroupCell, GroupCellTitle } from './GroupCell.styled';

describe('GroupCell', () => {
    it('should rennder properly', () => {
        const { container } = TestUtils.render(
            <GroupCell hide={false}>
                <Text>Dummy</Text>
            </GroupCell>
        );
        expect(container).toMatchSnapshot();
    });

    it('should not rennder hide prop is given', () => {
        const { container } = TestUtils.render(
            <GroupCell hide>
                <Text>Dummy</Text>
            </GroupCell>
        );
        expect(container).toMatchSnapshot();
    });

    it('should rennder GroupCellTitle properly', () => {
        const { container } = TestUtils.render(
            <GroupCellTitle>
                <Text>Dummy</Text>
            </GroupCellTitle>
        );
        expect(container).toMatchSnapshot();
    });
});
