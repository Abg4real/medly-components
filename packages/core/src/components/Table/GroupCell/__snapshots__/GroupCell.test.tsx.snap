// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupCell should not render hide prop is given 1`] = `
.c1 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
}

.c0 {
  display: grid;
  opacity: 0;
  width: 0px;
  height: 0px;
}

<div>
  <div
    class="c0"
  >
    <span
      class="c1"
    >
      Dummy
    </span>
  </div>
</div>
`;

exports[`GroupCell should render GroupCellTitle properly 1`] = `
.c1 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
}

.c0 {
  margin: 0;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: center;
  padding: 5px 0 5px 10px;
  grid-column: -1/1;
  text-overflow: initial;
  overflow: auto;
  white-space: normal;
  border-right: 1px solid #C7D0D8;
  border-bottom: 1px solid #C7D0D8;
}

<div>
  <span
    class="c0"
  >
    <span
      class="c1"
    >
      Dummy
    </span>
  </span>
</div>
`;

exports[`GroupCell should render properly 1`] = `
.c1 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
}

.c0 {
  display: grid;
  opacity: 1;
  width: 100%;
  height: 100%;
}

<div>
  <div
    class="c0"
  >
    <span
      class="c1"
    >
      Dummy
    </span>
  </div>
</div>
`;
